src/app/server/Mdns.cpp

```diff
@@ -48,7 +48,7 @@ NodeId GetCurrentNodeId()
 
     // Search for one admin pairing and use its node id.
     auto pairing = GetGlobalAdminPairingTable().cbegin();
-    if (pairing != GetGlobalAdminPairingTable().cend())
+    if (pairing != GetGlobalAdminPairingTable().cend() && pairing->GetNodeId() != kUndefinedNodeId)
```

Comment:
Wrt the TODO above:

`// TODO: once operational credentials are implemented, node ID should be read from them`

I would think that once we are getting the node ID from operational credentials, we should remove this check.